geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/BinAggregatingScan.scala

```diff
@@ -155,9 +141,43 @@ object BinAggregatingScan {
     (Seq(hints.getBinTrackIdField) ++ geom ++ dtg ++ hints.getBinLabelField).distinct.filter(_ != "id")
   }
 
-  // wrapper for java's byte buffer that adds scala methods for the aggregating iterator
-  class ByteBufferResult(val buffer: ByteBuffer, var overflow: ByteBuffer) {
-    def ensureCapacity(size: Int): ByteBuffer = {
+  class ResultCallback(buffer: ByteBuffer, private var overflow: ByteBuffer) extends BinaryOutputCallback {
```

Comment:
I'm tempted to suggest that this class name should include the word "Binary" for 'clarity'.  That said, anyone who can read the rest of this can probably figure it out.:)